### PR TITLE
Pre-emptive Auth with IT

### PR DESCRIPTION
**Pull request by mohideen**
https://jira.duraspace.org/browse/FCREPO-1761

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -48,6 +49,9 @@
     <!-- osgi -->
     <osgi.export.packages>org.fcrepo.client.*;version=${project.version}</osgi.export.packages>
     <osgi.import.packages>*</osgi.import.packages>
+    <grizzly.version>2.3.18</grizzly.version>
+    <spring.version>4.1.4.RELEASE</spring.version>
+    <logback.version>1.1.2</logback.version>
```
Comment (acoburn) [inline pom.xml]: very minor point here, but can you move these dependencies higher up in the list, so as to keep them separate from the OSGi definitions.

File: pom.xml [hunk 2]
```diff
@@ -105,20 +109,94 @@
...
+      <groupId>org.glassfish.jersey.core</groupId>
+      <artifactId>jersey-common</artifactId>
+      <version>${jersey.version}</version>
+      <scope>test</scope>
+    </dependency>
+
     <dependency>
-      <groupId>org.apache.jena</groupId>
-      <artifactId>jena-arq</artifactId>
-      <version>${jena.version}</version>
```
Comment (acoburn) [inline pom.xml]: As you're removing the jena dependency, can you also remove the `<jena.version/>` line in the properties above?

File: pom.xml [hunk 3]
```diff
@@ -188,5 +272,19 @@
...
+    <id>nexus</id>
+    <name>Releases</name>
+    <url>https://maven.lib.umd.edu/nexus/content/repositories/releases</url>
+  </repository>
+  <snapshotRepository>
+    <id>nexus</id>
+    <name>Snapshots</name>
+    <url>https://maven.lib.umd.edu/nexus/content/repositories/snapshots</url>
+  </snapshotRepository>
+</distributionManagement>
```
Comment (acoburn) [inline pom.xml]: This section should be removed -- the artifacts should not, by default, be pushed to a UMD maven repo.

**Comment by acoburn** [comment]
It's nice to see that you're using vim, but please remove the `src/main/java/org/fcrepo/client/.FcrepoHttpClientBuilder.java.swp` file from this PR.

File: src/test/resources/log4j.properties [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Root logger option
+log4j.logger.org.fcrepo=INFO,stdout
+#log4j.logger.org.fcrepo.client=DEBUG,stdout
+
+# Direct log messages to stdout
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+#log4j.appender.stdout.Target=System.out
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
```
Comment (acoburn) [inline src/test/resources/log4j.properties]: channeling @awoods here: please replace the log4j config with logback format. Here is an example: https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/test/resources/logback-test.xml

File: src/test/resources/spring-test/master.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:context="http://www.springframework.org/schema/context"
+  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-4.0.xsd
+    http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-3.0.xsd">
```
Comment (acoburn) [inline src/test/resources/spring-test/master.xml]: The Spring XSDs should be "versionless". You should be able to just remove the `-3.0` and `-4.0` portion of the strings.

File: src/test/resources/spring-test/repo.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xmlns:c="http://www.springframework.org/schema/c"
+       xmlns:util="http://www.springframework.org/schema/util"
+       xsi:schemaLocation="
+    http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-4.0.xsd
+    http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-4.0.xsd
```
Comment (acoburn) [inline src/test/resources/spring-test/repo.xml]: See above about "versionless" XSD locations

File: src/test/resources/spring-test/rest.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:context="http://www.springframework.org/schema/context"
+  xmlns:util="http://www.springframework.org/schema/util"
+  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
+    http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-3.0.xsd http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd">
```
Comment (acoburn) [inline src/test/resources/spring-test/rest.xml]: See above about versionless XSDs

File: src/test/resources/spring-test/test-container.xml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:context="http://www.springframework.org/schema/context"
+  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
+  http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-3.0.xsd">
```
Comment (acoburn) [inline src/test/resources/spring-test/test-container.xml]: Ditto.

File: src/test/resources/log4j.properties (hunk 4 above)
Comment (awoods) [inline src/test/resources/log4j.properties]: +1

**Comment by mohideen** [comment]
Working on merging the latest changes from master to resolve the conflict.

**Comment by mohideen** [comment]
I had to add the joda-time dependency as ConnectionManagementTest was failing with NoSuchMethod even in the master branch.

**Comment by awoods** [comment]
@mohideen, if this is ready for re-review, please move the JIRA ticket to "Start Review":
https://jira.duraspace.org/browse/FCREPO-1761

**Comment by acoburn** [comment]
Superseded by #9

**Pull request closed without merging**